resolver.go

```diff
@@ -151,22 +158,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
```

Comment:
My tests wouldn't have covered all the cases to verify the ordering rules of rfc 3484. So my claim about getaddrinfo() returning the v4 and v6 results in the same order it received from the server could be wrong. But my intent wasn't to test the getaddrinfo() implementation. Leaving aside the specifics of how geraddrindo() sorts the results, what is your contention with respect to this PR ?